nca/NetworkConfig/NetworkConfig.py

```diff
@@ -265,6 +268,28 @@ def allowed_connections(self, from_peer, to_peer, layer_name=None):
 
         return allowed_conns_res, captured_flag_res, allowed_captured_conns_res, denied_conns_res
 
+    def allowed_connections_optimized(self, layer_name=None):
```

Comment:
connectivity properties of peers which are HostEndpoints should be split and handled separately, as their connectivity is only determined by Calico layer.

Reply:
Changed according to the original implementation.